src/core/qgsvectorlayerutils.cpp

```diff
@@ -380,17 +380,24 @@ QgsFeature QgsVectorLayerUtils::createFeature( const QgsVectorLayer *layer, cons
     bool checkUnique = true;
 
     // in order of priority:
+    // 1. passed attribute value
+    if ( attributes.contains( idx )
+         && !( fields.at( idx ).constraints().constraints() & QgsFieldConstraints::ConstraintUnique ) )
+    {
+      v = attributes.value( idx );
+    }
```

Comment:
can we set `checkUnique = false;` here? Just to skip the last `if`

Reply:
Sure! No problem!